### PR TITLE
Feature tr 314 add channel type in navitia

### PR DESCRIPTION
**Pull request by kadhikari**
Channel types added for each channel.
For more information please refer to http://jira.canaltp.fr/browse/TR-314

### CODE REVIEW

**Comment by azime** [comment]
We must change the version of database

File: source/kraken/fill_disruption_from_database.h [hunk 1]
```diff
@@ -78,7 +81,7 @@ namespace navitia {
                 fill_disruption(const_it);
                 fill_cause(const_it);
                 tag = nullptr;
-                impact = nullptr;
+                impact = nullptr;
```
Comment (azime) [inline source/kraken/fill_disruption_from_database.h]: remove space

File: source/routing/tests/routing_api_test_data.h [hunk 2]
```diff
@@ -653,10 +659,11 @@ struct routing_api_data {
 
             impact->informed_entities.push_back(make_pt_obj(nt::Type_e::Line, "A", *b.data->pt_data, impact));
             impact->informed_entities.push_back(make_pt_obj(nt::Type_e::Line, "B", *b.data->pt_data, impact));
-            impact->informed_entities.push_back(make_pt_obj(nt::Type_e::Line, "C", *b.data->pt_data, impact));
+            impact->informed_entities.push_back(make_pt_obj(nt::Type_e::Line, "C", *b.data->pt_data, impact));
```
Comment (azime) [inline source/routing/tests/routing_api_test_data.h]: remove space

File: source/kraken/fill_disruption_from_database.h (hunk 1 above)
Comment (kadhikari) [inline source/kraken/fill_disruption_from_database.h]: ok

File: source/routing/tests/routing_api_test_data.h (hunk 2 above)
Comment (kadhikari) [inline source/routing/tests/routing_api_test_data.h]: ok

**Comment by kadhikari** [comment]
Abder, i did some modification on your remarks

File: source/jormungandr/jormungandr/interfaces/v1/fields.py [hunk 3]
```diff
@@ -225,6 +225,14 @@ def output(self, key, obj):
         enum = type_pb2._ACTIVESTATUS
         return str.lower(enum.values_by_number[status].name)
 
+class channel_types(fields.Raw):
```
Comment (antoine-de) [inline source/jormungandr/jormungandr/interfaces/v1/fields.py:228]: the `PBEnum` was not working ?

File: source/jormungandr/tests/chaos_disruptions_tests.py [hunk 4]
```diff
@@ -562,19 +582,23 @@ def make_mock_chaos_item(disruption_name, impacted_obj, impacted_obj_type, start
         pb_end.uri = end
         pb_end.pt_object_type = chaos_pb2.PtObject.stop_area
 
-    # Messages
+    # Message with one channel and two channel types: web and sms
```
Comment (antoine-de) [inline source/jormungandr/tests/chaos_disruptions_tests.py]: you only have sms here no ?

File: source/jormungandr/tests/disruptions_tests.py [hunk 5]
```diff
@@ -326,6 +326,23 @@ def test_disruption_on_route(self):
...
 
+        #Check message, channel and types
+        disruption_message = get_not_null(response, 'disruptions')
+        eq_(len(disruption_message), 1)
+        message = get_not_null(disruption_message[0], 'messages')
+        eq_(message[0]['text'], 'no luck')
+        channel = get_not_null(message[0], 'channel')
+        eq_(channel['id'], 'sms')
+        eq_(channel['name'], 'sms')
+        eq_(len(channel['types']), 2)
```
Comment (antoine-de) [inline source/jormungandr/tests/disruptions_tests.py]: you should also check the channel types no ?

File: source/jormungandr/tests/disruptions_tests.py [hunk 6]
```diff
@@ -326,6 +326,23 @@ def test_disruption_on_route(self):
...
+
+        eq_(message[1]['text'], 'try again')
+        channel = get_not_null(message[1], 'channel')
+        eq_(channel['id'], 'email')
+        eq_(channel['name'], 'email')
+        eq_(len(channel['types']), 2)
```
Comment (antoine-de) [inline source/jormungandr/tests/disruptions_tests.py]: same here

File: source/kraken/fill_disruption_from_chaos.cpp [hunk 7]
```diff
@@ -209,6 +209,39 @@ make_pt_objects(const google::protobuf::RepeatedPtrField<chaos::PtObject>& chaos
...
+            break;
+        case chaos::Channel_Type_twitter:
+            res.insert(nt::new_disruption::ChannelType::twitter);
+            break;
+        case chaos::Channel_Type_facebook:
+            res.insert(nt::new_disruption::ChannelType::facebook);
+            break;
+        case chaos::Channel_Type_unkown_type:
+            res.insert(nt::new_disruption::ChannelType::unknown_type);
+            break;
```
Comment (antoine-de) [inline source/kraken/fill_disruption_from_chaos.cpp:239]: you need a default with an error

File: source/routing/tests/routing_api_test_data.h [hunk 8]
```diff
@@ -682,8 +689,15 @@ struct routing_api_data {
 
             impact->informed_entities.push_back(make_pt_obj(nt::Type_e::Route, "A:0", *b.data->pt_data, impact));
 
-            impact->messages.push_back({"no luck", "sms", "sms", "content type", default_date, default_date});
-            impact->messages.push_back({"try again", "sms", "sms", "content type", default_date, default_date});
+            channel_types.clear();
+            channel_types.insert(ChannelType::web);
+            channel_types.insert(ChannelType::sms);
+            impact->messages.push_back({"no luck", "sms", "sms", "content type", default_date, default_date, channel_types});
```
Comment (antoine-de) [inline source/routing/tests/routing_api_test_data.h]: can't you do `impact->messages.push_back({"no luck", "sms", "sms", "content type", default_date, default_date, {ChannelType::web, ChannelType::sms}});`

File: source/kraken/fill_disruption_from_database.h [hunk 9]
```diff
@@ -252,5 +264,28 @@ namespace navitia {
             FILL_NULLABLE(channel, max_size, uint32_t)
         }
 
+        template<typename T>
+        void fill_channel_type(T const_it, chaos::Channel* channel) {
+            const auto& type_ = const_it["channel_type"].template as<std::string>();
+            //Here we have to verify existance of Channel Type in the channel.
+            if (type_ == "web"){
```
Comment (xlqian) [inline source/kraken/fill_disruption_from_database.h]: just for the consistence, can you use switch like you've done here https://github.com/CanalTP/navitia/pull/1168/files#diff-e609fca02cbf3693aa5cfeffb8a75685R213 ?

File: source/jormungandr/jormungandr/interfaces/v1/fields.py (hunk 3 above)
Comment (kadhikari) [inline source/jormungandr/jormungandr/interfaces/v1/fields.py:228]: I could not make it work

**Comment by kadhikari** [comment]
Antoine i modified except for PbNum

**Comment by TeXitoi** [comment]
This branch need a rebase.

**Comment by kadhikari** [comment]
Antoine, i rebased the latest version of navitia

**Pull request merged**